### ISSUE TITLE
Trouble connecting with Apple Calendar

### ISSUE THREAD

**Issue by jmfoleyjr** [Kozea/Radicale#954]
Is there something special that needs to be done for Apple Calendar? I have radicale clients working successfully on Linux, Windows and Mac using Thunderbird, but cannot get it working with Apple Calendar, even on the same computer where it does work with Thunderbird. I've run radicale in debug mode on the Mac for both Thunderbird and Apple Calendar. Results:

```
==> radical-labmacMacCal.out <==
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'http',
 'wsgi.version': (1, 0)}
[7f0a991ed700] DEBUG: Sanitized script name: ''
[7f0a991ed700] DEBUG: Sanitized path: '/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/'
[7f0a991ed700] INFO: Successful login: 'mark'
[7f0a991ed700] DEBUG: Rule 'mark':'mark' doesn't match '.+':'ConferenceCalendar(/.*)?' from section 'conference'
[7f0a991ed700] INFO: Rights: 'mark':'mark' doesn't match any section
[7f0a991ed700] WARNING: Access to principal path '/mark/' denied by rights backend
[7f0a991ed700] INFO: OPTIONS response status for '/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/' in 0.002 seconds: 200 OK

==> radical-labmacTbird.out <==

[7ff194a42700] DEBUG: Rule 'mark':'ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec' matches '.+':'ConferenceCalendar(/.*)?' from section 'conference'
[7ff194a42700] DEBUG: Client provided sync token: 'http://radicale.org/ns/sync/86549883de7ef4f4a9f44710b7a6362d'
[7ff194a42700] DEBUG: Response content:
<?xml version="1.0"?>
<multistatus xmlns="DAV:">
  <sync-token>http://radicale.org/ns/sync/86549883de7ef4f4a9f44710b7a6362d</sync-token>
</multistatus>

[7ff194a42700] INFO: REPORT response status for '/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/' with depth '1' in 0.005 seconds: 207 Multi-Status

```
Note that on Apple Calendar I get "Rule 'mark':'mark' doesn't match '.+':'ConferenceCalendar(/.*)?' from section 'conference'" whereas for Thunderbird I get, "DEBUG: Rule 'mark':'ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec' matches '.+':'ConferenceCalendar(/.*)?' from section 'conference'". Why?

To set up the account on the Mac, I selected, "Other CalDAV Account". I selected Account Type = "Advanced". I entered my user Id and password as configured in the htpasswd file. Server address = "host.mydom.org" (should it be https://host.mydom.org?). Server path = "/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/" -- same as with Thunderbird. Port = 5232. SSL checked. All these are exactly the same as Thunderbird settings as far as I know.

After doing this, the calendar name "host.mydom.org" does appear on the list of calendars, but no calendar appears. Instead, there is a small gray triangle beside the calendar name. Clicking on it gives the message:  The server specified an invalid calendar home. No calendar home was spcified for the account "host.mydom.org".

Does anyone have an idea of what could be wrong?

**Comment by jmfoleyjr** [comment]
No one has any insight on this? Here is some more information ...

Mac/Calendar still not working. Output of -D below. After configuring this calendar on Mac Calendar, the server host name (not the calendar name) appears on the Calendar app with a gray triangle. Clicking on that says, "The server specified an invalid calendar home. No calendar home was specified for the account "mail.ohprs.org".

My Mac/Calendar account setup is:
User Name: mark
Password: *******
Server Address: mail.ohprs.org
Server Path: /ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/
Port: 5232
Use SSL, checked
Use Kerberos, not checked

output from radicale -D:

[7f580e54a700] DEBUG: Sanitized script name: ''
[7f580e54a700] DEBUG: Sanitized path: '/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/'
[7f580e54a700] INFO: Successful login: 'mark'
[7f580e54a700] DEBUG: Rule 'mark':'mark' doesn't match '.+':'ConferenceCalendar(/.*)?' from section 'conference'
[7f580e54a700] INFO: Rights: 'mark':'mark' doesn't match any section
[7f580e54a700] WARNING: Access to principal path '/mark/' denied by rights backend
[7f580e54a700] INFO: OPTIONS response status for '/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/' in 0.003 seconds: 200 OK

I think it must have something to do with the rights file, which is: 

[conference]
user = .+
collection = ConferenceCalendar(/.*)?
permission = rw

This rights file works for Thunderbird on Linux, Mac and Windows 7, and for caldavsynchronizer with Outlook on Windows 7 and Windows 10. Only the Mac Calendar is giving me trouble. Any insight greatly appreciated (has anyone implement this with rights for Mac Calendar?). If I don't come up with a solution in the next week or so we will abandon the idea of using radicale for Mac Calendar clients.

**Comment by return42** [comment]
> Server Path: /ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/

My iOS & macOS experience is very old, but shouldn't it be:

    Server Path: /<user>/

**Comment by jmfoleyjr** [comment]
Well, clearly something is wrong, but not sure it's the server path. Thunderbird CalDav clients on Mac, Linux and Windows, and CalDavSynchronizer (Windows) connect just fine with the server path as shown. The actual Real path on the server is:

/home/radicale/collections/collection-root/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/

I don't see where <user> would fit in that path.

**Comment by return42** [comment]
No, you misunderstood me. What I mean is; the setup of the clients (e.g. thunderbird  and Apple's Calendar) are totally different .. while you have to enter the full path 

> My Mac/Calendar account setup is:
> User Name: mark
> Password: *******
> Server Address: mail.ohprs.org

this server path might be OK in thunderbird ..

> Server Path: /ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/

In Apple calendar it is simply 

> Server Path: /mark/
> Port: 5232
> Use SSL, checked
> Use Kerberos, not checked

this is what I have setup in Apples Calendar V 9.0

**Comment by return42** [comment]
Or even try:

> Server Path: /ConferenceCalendar

**Comment by jmfoleyjr** [comment]
Well, hallelujah! The /user/ path didn't work, but  /ConferenceCalendar/ did! I'll do some testing on this to make sure everything is working.

**Comment by jonassmedegaard** [comment]
Quoting jmfoleyjr (2019-05-30 15:05:59)
> Well, clearly something is wrong, but not sure it's the server path. Thunderbird CalDav clients on Mac, Linux and Windows, and CalDavSynchronizer (Windows) connect just fine with the server path as shown. The actual Real path on the server is:
> 
> /home/radicale/collections/collection-root/ConferenceCalendar/2c7a4b91-e0d9-6d05-d6ec-29112fec18ec/
> 
> I don't see where <user> would fit in that path.
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Kozea/Radicale/issues/954#issuecomment-497322342

CalDAV protocol supports elegant negotiations between client and server 
about which resources to get.

Crude simple old crappy CalDAV clients does not make use of those 
negotiation features, but need explicit paths.

Elegant modern smart CalDAV clients making use of negotiation features 
can therefore not always reach same resources as crude simple old crappy 
ones.

Correct, there is no "user" part in above URL which means you will 
likely have a hard time telling elegant CalDAV clients how to reach it.


 - Jonas

-- 
 * Jonas Smedegaard - idealist & Internet-arkitekt
 * Tlf.: +45 40843136  Website: http://dr.jones.dk/

 [x] quote me freely  [ ] ask before reusing  [ ] keep private
